recipe/meta.yaml

```
@@ -1,33 +1,34 @@
 {% set name = "libassuan" %}
 {% set version = "3.0.1" %}
-{% set sha256 = "c8f0f42e6103dea4b1a6a483cb556654e97302c7465308f58363778f95f194b1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
   url: https://www.gnupg.org/ftp/gcrypt/{{ name }}/{{ name }}-{{ version }}.tar.bz2
-  sha256: {{ sha256 }}
+  sha256: c8f0f42e6103dea4b1a6a483cb556654e97302c7465308f58363778f95f194b1
 
 build:
   number: 0
   skip: true  # [win]
+  # libgpg-error does not exist on s390x
+  skip: True  # [s390x]
   run_exports:
     - {{ pin_subpackage('libassuan', max_pin='x') }}
 
 requirements:
   build:
     - {{ compiler('c') }}
-    - {{ stdlib("c") }}
     - {{ compiler('cxx') }}
     - make       # [unix]
     - libtool    # [unix]
     - automake   # [unix]
     - gnuconfig  # [unix]
   host:
-    - libgpg-error
+    # https://github.com/gpg/libassuan/blob/libassuan-3.0.1/configure.ac#L402
+    # but avaliable only version 1.42 in the main channel.
+    - libgpg-error ==1.42
```

The min version of 1.17 (also in the [NEWS](https://github.com/gpg/libassuan/blob/libassuan-3.0.1/NEWS#L192) file from 9 years ago!) suggests:

```suggestion
    - libgpg-error >=1.17
```

---

@ifitchet host_section_needs_exact_pinnings: Linked libraries host should have exact version innings.
That is why I used ==1.42

---

the linter complains because of potential problems reproducing the build, especially when the host dependencies have ABI/API issues which is very common for lib<name>  dependencies

---

My bad, yes, fixed pinnings in `host`.